scout/adapter/mongo/event.py

```diff
@@ -79,26 +79,45 @@ def events(self, institute, case=None, variant_id=None, level=None,
               list: filtered query returning matching events
         """
 
-        query = {'institute': institute['_id']}
+        query = {}
 
         if variant_id:
-            query['category'] = 'variant'
-            query['variant_id'] = variant_id
-            if level:
-                query['level'] = level
-                if level != 'global':
-                    query['case'] = case['_id']
-        elif panel:
-            query['panel'] = panel
-        # If no variant_id or panel we know that it is a case level comment
+            if comments:
+                # If it's comment-related event collect global and variant-specific comment events
+                query = {
+                    '$or': [
+                        {
+                            'category' : 'variant',
+                            'variant_id' : variant_id,
+                            'verb' : 'comment',
+                            'level' : 'global'
+                        },
+                        {
+                            'category' : 'variant',
+                            'variant_id' : variant_id,
+                            'institute' : institute['_id'],
+                            'case' : case['_id'],
+                            'verb' : 'comment',
+                            'level' : 'specific'
+                        }
+                    ]
+                }
+            else: # Collect other variant-specific events which are not comments
+                query['institute'] = institute['_id']
+                query['category'] = 'variant'
+                query['variant_id'] = variant_id
+                query['case'] = case['_id']
         else:
-            query['category'] = 'case'
+            query['institute'] = institute['_id']
+            if panel:
+                query['panel'] = panel
+            # If no variant_id or panel we know that it is a case level comment
+            else:
+                query['category'] = 'case'
 
-            if case:
-                query['case'] = case['_id']
+                if case:
+                    query['case'] = case['_id']
 
-        if comments:
-            query['verb'] = 'comment'
 
         return self.event_collection.find(query).sort('created_at', pymongo.DESCENDING)
```

Comment:
@northwestwitch is this the query that takes to long when building the report?

Reply:
It might be, but we could be able to fix the issue by increasing the timeout of gunicorn. But if you feel like optimizing the query (by adding an index here?) please do!

Reply:
I will at least give it a thought:)

Reply:
This was not an issue for the creation of the case report. We solved that already, so this query is fine!